internal/controller/bucket/policy.go

```diff
@@ -0,0 +1,203 @@
+package bucket
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/smithy-go"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	"github.com/linode/provider-ceph/apis/provider-ceph/v1alpha1"
+	apisv1alpha1 "github.com/linode/provider-ceph/apis/v1alpha1"
+	"github.com/linode/provider-ceph/internal/backendstore"
+	"github.com/linode/provider-ceph/internal/consts"
+	"github.com/linode/provider-ceph/internal/controller/s3clienthandler"
+	"github.com/linode/provider-ceph/internal/otel/traces"
+	"github.com/linode/provider-ceph/internal/rgw"
+	"go.opentelemetry.io/otel"
+)
+
+// PolicyClient is the client for API methods and reconciling the BucketPolicy
+type BucketPolicyClient struct {
```

Comment:
I don't think we need the `Bucket` prefix here given we're within the `bucket` package - so I would rename this to `PolicyClient` to be consistent with `ACLClient` and `LifecycleConfigurationClient`

Reply:
Removed those prefixes in https://github.com/linode/provider-ceph/pull/233/commits/a0f1bddb3e36d0fd5ae192d78ad5085ed109be85
(Bucket type was also changed)